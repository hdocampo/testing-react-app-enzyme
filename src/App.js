import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

import ImageContainer from './components/ImageContainer/imageContainer'

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      appExtraClass: false
    }

    this.handleClickOnImg = this.handleClickOnImg.bind(this);
    this.handleClassesAfterClick = this.handleClassesAfterClick.bind(this);
  }

  handleClickOnImg() {
    this.setState({ appExtraClass: true })
  }

  handleClassesAfterClick() {
    this.handleClickOnImg();
  }

  render() {
    const appContainerClasses = `App ${this.state.appExtraClass ? 'App-clicked' : ''}`;

    const imgConfig = {
      className: 'App-logo',
      alt: 'logo',
      src: logo
    }
    return (
      <div className={appContainerClasses}>
        <header className="App-header">
          <ImageContainer
            onClick={this.handleClassesAfterClick}
            {...imgConfig}
          />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

export default App;
