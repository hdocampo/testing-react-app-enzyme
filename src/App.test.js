import React from 'react';
import { mount, shallow, render } from 'enzyme';

import App from './App';

describe('App', () => {
  it('should render the App component', () => {
    const component = shallow(<App />);

    expect(component).toMatchSnapshot();

    component.unmount();
  })
})