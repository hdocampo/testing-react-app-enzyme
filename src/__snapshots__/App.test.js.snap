// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component 1`] = `
<div
  className="App "
>
  <header
    className="App-header"
  >
    <ImageContainer
      alt="logo"
      className="App-logo"
      onClick={[Function]}
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
